### DISCUSSION TITLE
How to use Polyhedron_3 with new surface_Delaunay_remeshing method ?

### DISCUSSION THREAD

**Discussion by hanilan** [CGAL/cgal#7965] [Q&A] [answered]
Hi,
I try to use the new surface_Delaunay_remeshing() method provided in CGAL 5.6 with Polyhedron_3 but I struggle to compile my source as there are a lot of missing things when using Polyhedron_3 instead Surface_mesh. 
What I did:
-  Add a VertexItem derived from HalfedgeDS_vertex_max_base_with_id (need to add a time stamp and redefine 3 ctor)
![image](https://github.com/CGAL/cgal/assets/52419411/7da590a3-7302-4ded-8ab6-b7c31268b294)
- Add a patch_id() method in FaceItem
- Add a face_index_map, a vertex_point_map and a face_patch_map when calling surface_Delaunay_remeshing()

But I have still missing functions as set_feature_edge() for example. I can continue to add missing functions to achieve to compile my sample but I'm wondering if it is normal to not have a straightforward way to use Polyhedron_3 with this function. It seems Mesh_Polyhedron_3 is more suitable as it fill some requirements need by surface_Delaunay_remeshing() but I don't think you can override the polyhedron items. Maybe I miss something but it would be interesting to have an example of using a Polyhedron_3 with surface_Delaunay_remeshing() in CGAL doc.

**Comment by hanilan** [comment, marked as answer]
I finally succeed to compile in copying the polyhedron items code in Mesh_Polyhedron_3 to have my own items. But not an easy way to do the first time.

**Comment by afabri** [comment]
As the function only requires a model of `FaceGraph` I do not see why you need an extra items class. I will have a look.  If this is needed this is at least  a documentation bug.

**Reply by hanilan** [reply]
Yes I don't need extra items class to use the function but I meant if you need to have them you have to rewrite item code in Mesh_polyhedron_3 or maybe unherit from them. But soorry, my first message was unclear about that.

**Comment by afabri** [comment]
Hi @hanilan, Can you put your solution on gist.github.com, so that we can close the discussion. Best, Andreas

**Reply by hanilan** [reply]
Hi Andreas, the sample is here https://gist.github.com/hanilan/7c3fe21bdaf0fcb1995dbc80ce1b7f3d.
